test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,132 @@
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+
+
+@tags('cray', 'smoke')
+class TestCraySmokeTest(TestFunctional):
+
+    """
+    Set of tests that qualifies as smoketest for Cray platform
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    @tags('cray', 'smoke')
+    def test_cray_apstat(self):
+        """
+        Check output of apstat -rn on Cray.
+        The compute node summary should show that there are no nodes in use,
+        held, or down.  Check the that the number of up nodes is the same as
+        the number of available nodes.
+        """
+        cu = CrayUtils()
+        self.assertTrue(cu.count_node_in_state('resv') is 0,
+                        "No compute node should be having ALPS reservation")
+        self.assertTrue(cu.count_node_in_state('use') is 0,
+                        "No compute node should be in use")
+        nodes_up_apstat = cu.count_node_in_state('up')
+        nodes = self.server.filter(NODE,
+                                   {ATTR_rescavail+'.vntype': 'cray_compute'})
+        self.assertEqual(nodes_up_apstat,
+                         len(nodes[ATTR_rescavail+'.vntype=cray_compute']))
+
+    @tags('cray', 'smoke')
+    def test_cray_pbsnodes(self):
+        """
+        Check pbsnodes -av output on a Cray system.
+        pbsnodes output should show that nodes are free and resources are
+        available.
+        """
+        nodes = self.server.status(NODE)
+        for node in nodes:
+            self.assertEqual(node['state'], 'free')
+            self.assertEqual(node['resources_assigned.ncpus'], '0')
+            self.assertEqual(node['resources_assigned.mem'], '0kb')
+
+    @tags('cray', 'smoke')
+    def test_cray_login_job(self):
+        """
+        Submit a simple sleep job that requests to run on a login node
+        and expect that job to go in running state on a login node.
+        """
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_login'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a login node
+        qstat = self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = qstat[0]['exec_vnode'].partition(':')[0].strip('(')
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_login'},
+                           id=vname)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use are 0
+        self.assertEqual(cu.count_node_in_state('use'), 0)
+
```

Comment:
Please add a check that job ends normally and job output and error files are saved correctly.

Reply:
Sure